src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -394,9 +394,13 @@ private static boolean enumEqCheck(Object value, EnumDesc<?> label, MethodHandle
             Object resolved;
 
             try {
+                if (!(value instanceof Enum<?> enumValue)) {
```

Comment:
this code seems to be here with the only purpose of casting `value` to `Enum<?>` shouldn't this intention be more explicit?

Reply:
This method is also used in the `typeSwitch` (i.e. ordinary pattern matching switch, which may contain qualified enum constants intermixed with patterns), and so the `value` may be any object. If it is not an Enum, we can avoid the potential costly resolution.

Reply:
I see shouldn't we then have a test that stresses this code?

Reply:
I thought there's quite a few such tests, but turns out there's just a few (or maybe only one). So adding some more: [cbc9cb5](https://github.com/openjdk/jdk/pull/16499/commits/cbc9cb569b06c2f3d5e503519082c6243b51f9c8). Thanks!